### PR TITLE
Fix incorrect symbol occurrence for toLong/toInt primitive conversion

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes https://github.com/scalameta/scalameta/issues/1431

Previously, semanticdb-scalac emitted incorrect symbol occurrences whenever
primitive conversion methods were implicitly applied (for example, `scala/Int#toLong`).
This commit mostly fixes this bug, and adds a test case to demonstrate
for what corner cases the bug is still unfixed.

### CODE REVIEW

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala [hunk 1]
```diff
@@ -295,6 +295,29 @@ trait SymbolOps { self: SemanticdbOps =>
...
+      g.termNames.toShort,
+      g.termNames.toChar,
+      g.termNames.toInt,
+      g.termNames.toLong,
+      g.termNames.toFloat,
+      g.termNames.toDouble
+    )
+    // Returns true if the `this.sym` resolves to a primitive conversion method like toInt/toLong
+    // and the provided mtree name does not match the name of that symbol. This implementation
+    // has a minor bug where it doesn't detect implicit convesions when mtree.value has the name
```
Comment (kitbellew) [inline semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala:310]: conveRsions

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version=1.6.2
+sbt.version=1.5.8
```
Comment (kitbellew) [inline project/build.properties:1]: should this downgrade be done separately, if it is actually needed?

Comment (dos65) [reply project/build.properties:1]: not a perfect commit but let's take them together - every check for a possible release workflow fix takes about 1h.
if it will help - that's good.
anyway we will have to find out how to upgrade it the latest in future.

File: semanticdb/scalac/library/src/main/scala-2.13/scala/meta/internal/semanticdb/scalac/SymbolOpsCompat.scala [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    // Returns true if the `this.sym` resolves to a primitive conversion method like toInt/toLong
+    // and the provided mtree name does not match the name of that symbol. This implementation
+    // has a minor bug where it doesn't detect implicit convesions when mtree.value has the name
+    // of a conversion method like toInt/toLong, but it's a corner case that can be fixed separately
+    // if it's a big problem.
+    def isImplicitPrimitiveConversion(mtree: m.Name): Boolean = {
+      sym.name.startsWith("to") &&
+        g.definitions.ScalaValueClassesSet.contains(sym.owner) &&
+        implicitPrimitiveConversionNames.contains(sym.name.toTermName) &&
+        !sym.name.startsWith(mtree.value) &&
```
Comment (kitbellew) [inline semanticdb/scalac/library/src/main/scala-2.13/scala/meta/internal/semanticdb/scalac/SymbolOpsCompat.scala:26]: should this be `||`?

File: tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala [hunk 4]
```diff
@@ -205,4 +205,24 @@ class TargetedSuite extends SemanticdbSuite {
 
     }
   )
+
+  targeted(
```
Comment (olafurpg) [inline tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala:209]: I kept this test to ensure that it runs in 2.11/2.12/2.13. The  expect tests only run in 2.12

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala (hunk 1 above)
Comment (olafurpg) [reply semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala]: Fixed

File: project/build.properties (hunk 2 above)
Comment (olafurpg) [reply project/build.properties]: I upgraded to 1.6.2 for this PR while I'm re-using an older 1.5.8 sbt shell locally for edit/test.

File: semanticdb/scalac/library/src/main/scala-2.13/scala/meta/internal/semanticdb/scalac/SymbolOpsCompat.scala (hunk 3 above)
Comment (olafurpg) [reply semanticdb/scalac/library/src/main/scala-2.13/scala/meta/internal/semanticdb/scalac/SymbolOpsCompat.scala]: I refactored this condition into a simpler `sym.name.toString() != mtree.value`

**Review by tgodzik — APPROVED** [review]
LGTM!
